core/arch/x86_64/kernel/thread.c

```diff
@@ -390,6 +390,24 @@ void __nostackcheck thread_clr_boot_thread(void)
 	l->curr_thread = THREAD_ID_INVALID;
 }
 
+/* main tss */
+static tss_t system_tss;
+
+static void switch_tss(short int thread_id)
```

Comment:
For thread_id, uint8_t is enough. And it's not switch tss, switch_interrupt_stack is better.